Cleanup warnings

There are a number of outstanding `panel` warnings.
When building on CircleCI, the environment variable `CI=true` is set, which makes npm treat warnings as errors. This means that until they are resolved, all CI builds will fail.

```
!/bin/bash -eo pipefail
npm run-script build
npm info it worked if it ends with ok
npm info using npm@4.2.0
npm info using node@v7.10.1
npm info lifecycle panel@0.1.0~prebuild: panel@0.1.0
npm info lifecycle panel@0.1.0~build: panel@0.1.0

> panel@0.1.0 build /home/circleci/repo
> react-scripts build

Creating an optimized production build...

Treating warnings as errors because process.env.CI = true.
Most CI servers set it automatically.

Failed to compile.

./src/lib/apollo.js
  Line 4:  'defaultDataIdFromObject' is defined but never used  no-unused-vars
  Line 7:  '_' is defined but never used                        no-unused-vars

./src/components/Admin/Secrets/index.js
  Line 232:  Duplicate key 'isSecret'  no-dupe-keys

./src/components/Admin/Projects/index.js
  Line 8:    'Drawer' is defined but never used                 no-unused-vars
  Line 10:   'AppBar' is defined but never used                 no-unused-vars
  Line 12:   'Dialog' is defined but never used                 no-unused-vars
  Line 13:   'DialogActions' is defined but never used          no-unused-vars
  Line 14:   'DialogContent' is defined but never used          no-unused-vars
  Line 15:   'DialogContentText' is defined but never used      no-unused-vars
  Line 16:   'DialogTitle' is defined but never used            no-unused-vars
  Line 18:   'InputField' is defined but never used             no-unused-vars
  Line 24:   'validatorjs' is defined but never used            no-unused-vars
  Line 25:   'MobxReactForm' is defined but never used          no-unused-vars
  Line 28:   'check' is defined but never used                  no-unused-vars
  Line 232:  'projects' is assigned a value but never used      no-unused-vars
  Line 237:  Expected to return a value in function             array-callback-return
  Line 243:  Expected to return a value in function             array-callback-return
  Line 268:  Expected to return a value in function             array-callback-return
  Line 312:  Expected to return a value in function             array-callback-return
  Line 314:  Expected to return a value in function             array-callback-return
  Line 319:  Expected a default case                            default-case
  Line 473:  Expected a default case                            default-case
  Line 485:  Expected to return a value in function             array-callback-return
  Line 487:  Expected a default case                            default-case
  Line 550:  Expected to return a value at the end of function  array-callback-return

./src/components/Project/Releases/index.js
  Line 52:   Unreachable code         no-unreachable
  Line 55:   Unreachable code         no-unreachable
  Line 58:   Unreachable code         no-unreachable
  Line 589:  Expected a default case  default-case



npm info lifecycle panel@0.1.0~build: Failed to exec build script
npm ERR! Linux 4.4.0-124-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run-script" "build"
npm ERR! node v7.10.1
npm ERR! npm  v4.2.0
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! panel@0.1.0 build: `react-scripts build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the panel@0.1.0 build script 'react-scripts build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the panel package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     react-scripts build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs panel
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls panel
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/circleci/.npm/_logs/2018-05-22T04_15_49_270Z-debug.log
Exited with code 1
```

--- a/src/components/Admin/Projects/index.js
+++ b/src/components/Admin/Projects/index.js
@@ -5,27 +5,15 @@ import Toolbar from 'material-ui/Toolbar';
 import Button from 'material-ui/Button';
 import Table, { TableCell, TableHead, TableBody, TableRow } from 'material-ui/Table';
 import Paper from 'material-ui/Paper';
-import Drawer from 'material-ui/Drawer';
 import Card, { CardContent } from 'material-ui/Card';
-import AppBar from 'material-ui/AppBar';
 import Link from 'react-router-dom/Link';
-import Dialog, {
-  DialogActions,
-  DialogContent,
-  DialogContentText,
-  DialogTitle,
-} from 'material-ui/Dialog';
-import InputField from 'components/Form/input-field';
 import Checkbox from 'material-ui/Checkbox';
 import { FormControlLabel } from 'material-ui/Form';
 import Loading from 'components/Utils/Loading';
 import styles from './style.module.css';
 import { observer, inject } from 'mobx-react';
-import validatorjs from 'validatorjs';
-import MobxReactForm from 'mobx-react-form';
 import { graphql } from 'react-apollo';
 import gql from 'graphql-tag';
-import { check } from 'graphql-anywhere';
 import _ from 'lodash';
 
 @graphql(gql`
@@ -229,18 +217,18 @@ export default class Projects extends React.Component {
   }
 
   onBatchDeploy(route53Deploy){
-    const { environments, projects } = this.props.data;
+    const { environments } = this.props.data;
 
     var self = this
     this.state.checkedProjects.forEach(function(projectID){
       let project = {}
-      self.props.data.projects.map(function(tmpProject){
+      self.props.data.projects.forEach(function(tmpProject){
         if(tmpProject.id === projectID) {
           project = tmpProject
         }
       })
 
-      project.environments.map(function(env){
+      project.environments.forEach(function(env){
         let _environment = _.find(environments, {id: env.id})
         let _project = _.find(_environment.projects, {id: project.id})
 
@@ -265,7 +253,7 @@ export default class Projects extends React.Component {
           }).then(({data}) => {
             // find checked extensions for that env
             self.props.data.refetch()
-            _project.extensions.map(function(projectExtension){
+            _project.extensions.forEach(function(projectExtension){
               if(self.state.checkedExtensions.includes(projectExtension.extension.id)) {
                 console.log('updating project extension ' + projectExtension.extension.name)
                 console.log({
@@ -309,9 +297,9 @@ export default class Projects extends React.Component {
     var completeReleases = 0
     var failedReleases = 0
 
-    environments.map(function(env){
+    environments.forEach(function(env){
       let _environment = _.find(environments, { id: env.id })
-      projects.map(function(project){
+      projects.forEach(function(project){
         let _project = _.find(_environment.projects, { id: project.id })
         console.log(_project)
         if(_project !== undefined) {
@@ -328,6 +316,8 @@ export default class Projects extends React.Component {
                 break;
               case "fetching":
                 runningReleases += 1
+                break;
+              default:
                 break;
             }
           }
@@ -462,7 +452,7 @@ export default class Projects extends React.Component {
                       </Link>
                     </TableCell>
                     <TableCell>
-                      {project.environments.map(function(env){
+                      {project.environments.forEach(function(env){
                         // get env in environments query
                         let _environment = _.find(environments, { id: env.id })
                         let _project = _.find(_environment.projects, { id: project.id })
@@ -480,9 +470,11 @@ export default class Projects extends React.Component {
                             case "failed":
                               color = "red"
                               break;
+                            default:
+                              break;
                           }
 
-                          _project.releases[0].releaseExtensions.map(function(releaseExtension){
+                          _project.releases[0].releaseExtensions.forEach(function(releaseExtension){
                             let status = "lightgray"
                             switch(releaseExtension.state){
                               case "complete":
@@ -496,6 +488,8 @@ export default class Projects extends React.Component {
                                 break;                                                              
                               case "failed":
                                 status = "red"
+                                break;
+                              default:
                                 break;
                             }                  
                             extensionStatuses.push(
@@ -547,7 +541,7 @@ export default class Projects extends React.Component {
                       <b>Available Extensions</b>
                     </Typography>
                     <Grid item xs={12}>
-                      {extensions.map(function(extension){
+                      {extensions.forEach(function(extension){
                         if(extension.environment.id === env.id){
                           return (
                             <FormControlLabel

--- a/src/components/Admin/Secrets/index.js
+++ b/src/components/Admin/Secrets/index.js
@@ -229,7 +229,6 @@ export default class Secrets extends React.Component {
         'isSecret': secret.isSecret,
         'environmentID': this.props.store.app.currentEnvironment.id,
         'scope': secret.scope,
-        'isSecret': secret.isSecret,
       })
 
       this.form.$('key').set('disabled', true)

--- a/src/components/Project/Releases/index.js
+++ b/src/components/Project/Releases/index.js
@@ -54,13 +54,10 @@ class ReleaseView extends React.Component {
           switch(release.releaseExtensions[i].state){  
             case "waiting":
               return (<Chip label={release.releaseExtensions[i].extension.extension.name} style={{ backgroundColor: "yellow", color: "black", marginRight: 4 }} />)
-            break;
             case "complete":
               return (<Chip label={release.releaseExtensions[i].extension.extension.name} style={{ backgroundColor: "green", color: "white", marginRight: 4 }} />)
-            break;
             case "failed":
               return (<Chip label={release.releaseExtensions[i].extension.extension.name} style={{ backgroundColor: "red", color: "white", marginRight: 4 }} />)               
-            break;
             default:
               return (<Chip label={release.releaseExtensions[i].extension.extension.name} style={{ backgroundColor: "yellow", color: "black", marginRight: 4 }} />)
           }
@@ -607,6 +604,8 @@ export default class Releases extends React.Component {
           baseGitUrl = baseGitUrl[0] + "/commit/"
         }
       break;
+      default:
+        break;
     }
 
     return (

--- a/src/lib/apollo.js
+++ b/src/lib/apollo.js
@@ -1,10 +1,9 @@
 import { ApolloClient } from 'apollo-client';
 import { createHttpLink } from 'apollo-link-http';
 import { setContext } from 'apollo-link-context';
-import { InMemoryCache, defaultDataIdFromObject } from 'apollo-cache-inmemory';
+import { InMemoryCache } from 'apollo-cache-inmemory';
 import { ApolloLink } from 'apollo-link';
 import { onError } from "apollo-link-error";
-import _ from 'lodash';
 
 export default (GRAPHQL_URI = process.env.REACT_APP_CIRCUIT_URI + '/query') => {
   const httpLink = createHttpLink({

